### PR TITLE
finish task 5

### PR DESCRIPTION
**Pull request by XingWei0920**
This is only for task 5 to view an individual article. My plan is to display corresponding comments plus the functionality to add a comment together with this page showing the individual article

### CODE REVIEW

File: components/ArticleProfile.jsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+const UserProfile=()=>{
+    const {articles,setArticles}=useContext(ArticlesContext)
+
+    const {article_id}=useParams()
+    const [article, setArticle]=useState([])
+    const [isLoading, setIsLoading]=useState(true)
+ 
+    useEffect(()=>{
+        fetch(`https://xing-wei-app.onrender.com/api/articles/${article_id}`)
```
Comment (nc-poppy-m) [inline components/ArticleProfile.jsx]: I strongly recommend using axios over fetch - fetch will become more difficult to use as you get into queries etc. https://notes.northcoders.com/courses/js-front-end/axios

File: components/ArticleProfile.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import {useParams} from "react-router-dom"
+import {useState, useEffect, useContext} from "react"
+import { ArticlesContext } from "../contexts/ArticlesContext"
+import ArticleCard from "./ArticleCard"
+
+
+const UserProfile=()=>{
```
Comment (nc-poppy-m) [inline components/ArticleProfile.jsx]: What is this UserProfile component?

File: components/ArticleCard.jsx [hunk 3]
```diff
@@ -3,9 +3,10 @@ import {Link} from 'react-router-dom'
 
 const ArticleCard=(props)=>{
     const {article}=props
+    console.log(article)
```
Comment (nc-poppy-m) [inline components/ArticleCard.jsx]: Remember to remove logs

File: components/ArticleProfile.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+     if (isLoading)
+     {
+        return <h2>Loading ...</h2>
+     }
+
+   
+
+    return (
+        <>
+        <h2 className="itemBlock">
```
Comment (nc-poppy-m) [inline components/ArticleProfile.jsx:31]: If this is your article page, it needs to have to body

**Review by nc-poppy-m — CHANGES_REQUESTED** [review]
Good work so far! There are a couple of things left to check:
- Your single article needs a 'body' and should ideally be styled differently to your articleCard. The styling doesn't need to be all there yet, just an indication that it's different to your articleCard.
- Your UserProfile component name is a little bit hard to understand, please change it to something more semantic
- use Axios and separate it into a separate api file
- remove console.logs

Optional (but to consider):
- Looking at your plan, I'm not sure that articles needs to be in a whole react context given that you only need it for a few pages

Act on the feedback above and then resubmit when ready

**Review by popatre — CHANGES_REQUESTED** [review]
Nice refactor to axios. 

As Poppy suggested:

- [ ] There still doesnt seem to be a body on the article. I can't actually read what the article is because of this, which being an article, is the main point of it. 
- [ ] Im not sure what the articleContext is, or why its needed. Are there lots of pages in the app that are going to need this? If not, I'd suggest it isnt actually required

**Review by nc-modou — COMMENTED** [review]
Good response to feedback, feel free to merge this task.

One thing to note in general about the site, all of your data shown about the article has a label like this...
**article** : 
But if the user is on an article page, it should be clear that they're looking at data relating to an article, so I'd rename stuff to be a bit shorter like
Author:
Topic:
Votes: 0

**Pull request merged**